### PR TITLE
feat(flavors): add flavors - INNO-927

### PR DESCRIPTION
**Pull request by kalinchernev**
Because this pull request involved many changes, a detailed summary of changes in the form of documentation can be found in the file `05-flavors.md`, which can be seen, for example, at: https://github.com/ec-europa/europa-component-library/blob/cd4a4b1c1b30aff59eb25e66f9daa27e946c5d58/packages/styleguides/docs/src/05-flavors.md

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit ddd7d31a8c23b0dd51cc98cac8e447b15e548c60

https://deploy-preview-567--europa-component-library.netlify.com

**Comment by kalinchernev** [comment]
Script for generating a combined style guide on github pages is in progress

**Comment by kalinchernev** [comment]
Blocked by https://github.com/ec-europa/ecl-toolkit/pull/59 which should be merged and theme should be published and feeded in here in the old way before going forward.

File: .eslintignore [hunk 1]
```diff
@@ -1,6 +1,9 @@
-/dist
-/static/framework
+dist
 node_modules
 
 # this is only for prettier (atom extension doesn't support .prettierignore)
 package.json
+
+# Temporary build files of the style guides
+packages/styleguides/ec/static/scripts
```
Comment (yhuard) [inline .eslintignore]: Why here and not in the sub `packages/styleguides/*/.gitignore` files?

File: scripts/styleguide.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+cd packages/styleguides/docs
+npx fractal build && npx cross-env NODE_ENV=production
```
Comment (yhuard) [inline scripts/styleguide.sh]: Why `npx cross-env NODE_ENV=production`?

File: .eslintignore (hunk 1 above)
Comment (kalinchernev) [reply .eslintignore]: > Only one .eslintignore file can be used at a time, so .eslintignore files other than the one in the current working directory will not be used.

I tried, but it appeared the eslint doesn't work as gitignore: https://eslint.org/docs/2.0.0/user-guide/configuring#ignoring-files-and-directories

Comment (yhuard) [reply .eslintignore]: I didn't catch it was `.eslintignore` and not `.gitignore` 😅 
You're right, there should only be one `.eslintignore` but gitignore's syntax is still valid (i.e. you could use `packages/styleguides/*/static/scripts`)

Comment (kalinchernev) [reply .eslintignore]: Alright, I used this approach to save a line, thanks

File: packages/styleguides/docs/src/05.flavors.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+### Maintaining flavors (Important notes on APIs)
+
+The split of the project introduces several important rules which should be followed in order to keep ECL project repository maintainable for contributors, and also easy to integrate by consumers of ECL builds and resources.
+
+#### Naming
+
+Each flavor should be named in a distinguishable manner:
+
+* In `package.json` one would have `@ec-europa/eu-logos`. `@ec-europa` namespace should be preserved for correct organization on [npmjs](https://www.npmjs.com/). The `eu-logos` is for the EU flavor, whereas `ec-logos` should be the one for the EC flavor.
```
Comment (yhuard) [inline packages/styleguides/docs/src/05.flavors.md]: I'm not sure that it's the best way to name packages. `@ec-europa/eu-logos` doesn't tell you the package is built within the ECL. I suggest `@ec-europa/ecl-logos-eu`, `@ec-europa/ecl-logos--eu` or, shorter,  `@ecl/eu-logos` or `@ecl/logos-eu`. Feel free to add more suggestions and we, as a team, will decide what naming convention to follow.

ping @degliwe @raduchiriac @emeryro

File: packages/styleguides/docs/src/05.flavors.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+* Components' files are to be named with `eu-` or `ec-` based on flavors: `eu-logos.twig`, `eu-logos.scss`, `eu-logos.config.js`, etc. This makes flavor-specific code to be easily discoverable in search, clear and simple enough, following existing patterns. (no underscores and suffixes)
+* Twig templates should be prefixed with `@ec-europa/` in order to be consistently consumable by [Europa Component Library Twig loader](https://github.com/openeuropa/ecl-twig-loader)
+
+Please note that all current EC-flavor components have preserved `@ec-europa/ecl-*` naming in order to give time for consumers of ECL components to plan changes based on flavors. The `@ec-europa/ecl-` namespace, in future, should be the place to store reusable parts of components such as mixins and variables.
+
+Also, for keeping current consumption of ECL releases smooth for users, releases are to include `framework.zip` with this structure:
```
Comment (yhuard) [inline packages/styleguides/docs/src/05.flavors.md]: Missing `framework` root folder, on purpose? When I unzip an archived named `framework.zip`, my first expectation is to get a `framework` folder (currently the case)

Comment (kalinchernev) [reply packages/styleguides/docs/src/05.flavors.md]: I will remove this part, no worries

**Pull request merged**